### PR TITLE
Support Restapi/gnmi control plane acls

### PR DESCRIPTION
**Pull request by ZhaohuiS**
Currently Sonic supports the following ACL tables.

IPV6_SNMP_ACL
EVERFLOW
SSH_ONLY
..

For the Restapi/gnmi use-cases, Sonic has to support a new Table: EXTERNAL_CLIENT of type CTRLPLANE, stage ingress

This shall match on 'src ip prefix' and dst port '8080'. Caclmgrd must parse this from acl.json and install as in the below example:

iptables -A INPUT -s 20.20.20.20/27 -p tcp --dport 8080 -j ACCEPT

or ip6tables if the 'src ip prefix' is IPv6.
**Steps:**

1. add EXTERNAL_CLIENT_ACL in config_db.json
```
"ACL_TABLE": {
        "NTP_ACL": {
            "policy_desc": "NTP_ACL",
            "services": [
                "NTP"
            ],
            "stage": "ingress",
            "type": "CTRLPLANE"
        },
        ...
       "EXTERNAL_CLIENT_ACL": {
            "policy_desc": "EXTERNAL_CLIENT_ACL",
            "services": [
                "EXTERNAL_CLIENT"
            ],
            "stage": "ingress",
            "type": "CTRLPLANE"
        }
}
```
2. config reload, then check show acl tables:
```
EXTERNAL_CLIENT_ACL  CTRLPLANE  EXTERNAL_CLIENT  EXTERNAL_CLIENT_ACL  ingress
NTP_ACL              CTRLPLANE  NTP              NTP_ACL              ingress
SNMP_ACL             CTRLPLANE  SNMP             SNMP_ACL             ingress
SSH_ONLY             CTRLPLANE  SSH              SSH_ONLY             ingress
```
3. load acl rules:
`acl-loader update full external_acl.json`
external_acl.json looks like this:

```
{
    "acl": {
        "acl-sets": {
            "acl-set": {
                "EXTERNAL_CLIENT-ACL": {
                    "acl-entries": {
                        "acl-entry": {
                            "1": {
                                "ip": {
                                    "config": {
                                        "source-ip-address": "20.0.0.3/32"
                                    }
                                }, 
                                "transport": {
				    "config": {
					"destination-port": "8081"
				    }
				},
                                "config": {
                                    "sequence-id": 1
                                }, 
                                "actions": {
                                    "config": {
                                        "forwarding-action": "ACCEPT"
                                    }
                                }
                            }
                        }
                    }, 
                    "config": {
                        "name": "EXTERNAL_CLIENT-ACL"
                    }
                }
            }
        }
    }
}
```

4. Check acl rules 
IPv4:
```
admin@vlab-03:~$ show acl rule
Table                Rule          Priority    Action    Match
-------------------  ------------  ----------  --------  -------------------
EXTERNAL_CLIENT_ACL  RULE_0        10000       ACCEPT    L4_DST_PORT: 8081
                                                         SRC_IP: 20.0.0.3/32
EXTERNAL_CLIENT_ACL  DEFAULT_RULE  1           DROP      ETHER_TYPE: 2048
```
IPv6:
```
admin@vlab-03:~$ show acl rule
Table                Rule          Priority    Action    Match
-------------------  ------------  ----------  --------  --------------------
EXTERNAL_CLIENT_ACL  RULE_0        10000       ACCEPT    L4_DST_PORT: 8081
                                                         SRC_IPV6: 2001::3/128
EXTERNAL_CLIENT_ACL  DEFAULT_RULE  1           DROP      ETHER_TYPE: 2048
```
5.  check iptables
IPv4:
```
-A INPUT -s 20.0.0.3/32 -p tcp -m tcp --dport 8081 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 8080 -j DROP
```
IPv6:
```
-A INPUT -s 2001::3/128 -p tcp -m tcp --dport 8081 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 8081 -j DROP
```

7. Also support dst port range

```
				"transport": {
				    "config": {
					"destination-port": "8081..8083"
				    }
				}
```
```
admin@vlab-03:~$ show acl rule
Table                Rule          Priority    Action    Match
-------------------  ------------  ----------  --------  ----------------------------
EXTERNAL_CLIENT_ACL  RULE_0        10000       ACCEPT    L4_DST_PORT_RANGE: 8081-8083
                                                         SRC_IP: 20.0.0.3/32
EXTERNAL_CLIENT_ACL  DEFAULT_RULE  1           DROP      ETHER_TYPE: 2048
```

```
-A INPUT -s 20.0.0.3/32 -p tcp -m tcp --dport 8081 -j ACCEPT
-A INPUT -s 20.0.0.3/32 -p tcp -m tcp --dport 8082 -j ACCEPT
-A INPUT -s 20.0.0.3/32 -p tcp -m tcp --dport 8083 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 8081 -j DROP
-A INPUT -p tcp -m tcp --dport 8082 -j DROP
-A INPUT -p tcp -m tcp --dport 8083 -j DROP
```

Signed-off-by: Zhaohui Sun <zhaohuisun@microsoft.com>

### CODE REVIEW

File: scripts/caclmgrd [hunk 1]
```diff
@@ -80,6 +80,11 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
             "dst_ports": ["22"],
             "multi_asic_ns_to_host_fwd":True
         },
+        "EXTERNAL_CLIENT": {
+            "ip_protocols": ["tcp"],
+            "dst_ports": ["8081"],
```
Comment (prsunny) [inline scripts/caclmgrd:85]: Let this be triggered via config

**Review by prsunny — APPROVED** [review]
lgtm, lets add some unit tests

**Comment by ZhaohuiS** [comment]
> lgtm, lets add some unit tests

@prsunny Sure, will do.

**Comment by prsunny** [comment]
> > lgtm, lets add some unit tests
> 
> @prsunny Sure, will do.

Thanks @ZhaohuiS , please add test for IPv6 as well

**Comment by ZhaohuiS** [comment]
> > > lgtm, lets add some unit tests
> > 
> > 
> > @prsunny Sure, will do.
> 
> Thanks @ZhaohuiS , please add test for IPv6 as well

@prsunny Hi Prince, I added 4 unit test cases which cover single port and port range for both ipv4 and ipv6 in my last commit.
Please help to review it again. Thanks!

**Comment by prsunny** [comment]
@ZhaohuiS , we need to get this for 202205. Can you please have the submodule update for 202205?

**Comment by ZhaohuiS** [comment]
> @ZhaohuiS , we need to get this for 202205. Can you please have the submodule update for 202205?

@prsunny Sure, will also add print warning log if dst_port is not defined.

**Comment by ZhaohuiS** [comment]
> @ZhaohuiS , we need to get this for 202205. Can you please have the submodule update for 202205?

@prsunny I submitted in https://github.com/sonic-net/sonic-buildimage/pull/11962 for 202205 branch. Please review, thanks.

**Pull request merged**